cnf-certification-test/lifecycle/suite.go

```diff
@@ -475,13 +522,16 @@ func testHighAvailability(env *provider.TestEnvironment) {
 
 		if st.Spec.Template.Spec.Affinity == nil ||
 			st.Spec.Template.Spec.Affinity.PodAntiAffinity == nil {
-			badStatefulSet = append(badStatefulSet, st.ToString())
 			tnf.ClaimFilePrintf("StatefulSet found without valid high availability: %s", st.ToString())
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewStatefulSetReportObject(st.Namespace, st.Name, "StatefulSet found without valid high availability", false))
+		}
```

Comment:
An `else` branch here might avoid the use of the `compliantStatefulSet` var.

Reply:
Done.